improve: always show full screen buttons and remember last state

ZookaOnGit opened the pull request:
<!-- Keep the title short & concise so anyone non-technical can understand it,
     the title appears in PTB changelogs -->
#### Brief overview of PR changes/additions
Remove small screen option from preferences and always show full screen buttons.  Remember last state upon reopening Mudlet (full screen or not).

#### Motivation for adding to Mudlet
Remove UI clutter.  More intuitive for users (just save the previous state).

#### Other info (issues closed, discussion etc)
This PR has evolved from my #7749 to remove an unnecessary preferences and includes @SlySven 's #7753 fixes which improved the functionality of fullscreen.

This actually doesn't work on KDE.  KDE always overrides to non full screen startup as tested with other applications (firefox, consoles/terminals)

ZookaOnGit committed:
improve: always show full screen buttons and remember last state

SlySven commented:
> This actually doesn't work on KDE. KDE always overrides to non full screen startup as tested with other applications (firefox, consoles/terminals)

I can reproduce the bug you describe - this machine does have two monitors (at present) AND an NVidia graphics card AND KDE Plasma as the desktop. Seems there are some workarounds but no definitive fix: https://bugs.kde.org/show_bug.cgi?id=427920

SlySven commented:
Appropos my last post I have found a DE settings, application placement on monitors where #7753 - (which I've updated from the development branch so it doesn't crash if the preferences are opened without a profile loaded) works as designed and goes full screen on start-up if that option is set. However this PR at present, if run in the same manner does NOT restore to Full-screen on start-up - even if it was closed (which reminds me I have to implement the built in <Cmd>+<Q> MacOS "Close application" menu action that Qt provides automagically on that OS to the others that don't) whilst running Full-screen.

As such it isn't doing what it says on the tin about "and remember last state".

vadi2 commented:
What are you thinking of exactly?

ZookaOnGit commented:
If you can just do the preferences changes (which will happen regardless) in your PR I will close this.

SlySven committed:
Fix: combine the handling of maximised and full-screen states

It seems that they were interfering with each other!

The previous option `"enableFullScreenMode"` has been removed as it is not
applicable to what is now stored - it will be replaced by `"fullScreen"`.

As there is no need to only create (or in the case of #7753 destroy) the
full-screen controls as per a user setting the code can be simplified and
done in the main `(void) mudlet::init()` method alongside the other
controls.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
It is simpler to fix your code - so review and merge this PR https://github.com/ZookaOnGit/Mudlet/pull/2 in your repo in the code here...

It seems we are also tracking the maximized state and handling them separately when they are both always in play as they will be now, makes things more complicated than they need to be - the above PR does them both together and seems to work for me. Mudlet now always starts up in the state it is left. (First time run on a new user's system will be "normal" - not maximised and not full-screen)...

ZookaOnGit committed:
Merge pull request #2 from SlySven/Fix_fullscreen-options

Fix: combine the handling of maximised and full-screen states

SlySven commented:
Okay, things are acceptable (although I'm not a fan of the overall change) as it should now correctly remember the "normal"/"maximised"/"full-screen" state ***between*** sessions (DE/OS permitting! :smirk:). However, I'll leave it to Vadi to confirm and approve this PR...

vadi2 commented:
Thanks, and if the players don't like it, we'll listen to that feedback.
